sdk/include/opentelemetry/sdk/trace/simple_processor.h

```diff
@@ -48,11 +57,29 @@ class SimpleSpanProcessor : public SpanProcessor
 
   void Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept override
   {
+    WaitLock();
     exporter_->Shutdown(timeout);
+    ClearLock();
   }
 
 private:
   std::unique_ptr<SpanExporter> exporter_;
+  std::atomic_flag lock_;
+
+  void WaitLock() noexcept
+  {
+    /* Note: We expect calls to exporter to be "fast", i.e. we do NOT incrementally back-off
+     * and wait/notify here, we just loop until we have access, then try again.
+     *
+     * This has the downside that we could be spinning a long time if the exporter is slow.
+     * Note: in C++20x we could use `.wait` to make this slightly better. This should move to
+     * an exponential-back-off / wait strategy.
```

Comment:
Having this moved into a separate `lock_guard` like class, we can easily optimize and use the `.wait` for C++20x.

Reply:
Agreed!  Maybe someone else will have time to optimise in the meantime too.